storm-core/src/jvm/org/apache/storm/daemon/worker/Worker.java

```diff
@@ -0,0 +1,426 @@
+        state.cleanup(60); // this is just in case supervisor is down so that disk doesn't fill up.
+        // it shouldn't take supervisor 120 seconds between listing dir and reading it
+    }
+
+    public void doExecutorHeartbeats() {
+        Map<List<Integer>, ExecutorStats> stats;
+        List<IRunningExecutor> executors = this.executorsAtom.get();
+        if (null == executors) {
+            stats = StatsUtil.mkEmptyExecutorZkHbs(workerState.executors);
+        } else {
+            stats = StatsUtil.convertExecutorZkHbs(executors.stream().collect(Collectors
+                .toMap((Function<IRunningExecutor, List<Long>>) IRunningExecutor::getExecutorId,
+                    (Function<IRunningExecutor, ExecutorStats>) IRunningExecutor::renderStats)));
+        }
+        Map<String, Object> zkHB = StatsUtil.mkZkWorkerHb(workerState.topologyId, stats, workerState.uptime.upTime());
+        try {
+            workerState.stormClusterState
+                .workerHeartbeat(workerState.topologyId, workerState.assignmentId, (long) workerState.port,
+                    StatsUtil.thriftifyZkWorkerHb(zkHB));
+        } catch (Exception ex) {
+            LOG.error("Worker failed to write heartbeats to ZK or Pacemaker...will retry", ex);
+        }
+    }
+
+    public void checkCredentialsChanged() {
+        Credentials newCreds = workerState.stormClusterState.credentials(topologyId, null);
+        if (! ObjectUtils.equals(newCreds, credentialsAtom.get())) {
+            // This does not have to be atomic, worst case we update when one is not needed
+            AuthUtils.updateSubject(subject, autoCreds, (null == newCreds) ? null : newCreds.get_creds());
+            for (IRunningExecutor executor : executorsAtom.get()) {
+                executor.credenetialsChanged(newCreds);
+            }
+            credentialsAtom.set(newCreds);
+        }
+    }
+
+    public void checkThrottleChanged() {
+        boolean throttleOn = workerState.stormClusterState.topologyBackpressure(topologyId, this::checkThrottleChanged);
+        workerState.throttleOn.set(throttleOn);
+    }
+
+    public void checkLogConfigChanged() {
+        LogConfig logConfig = workerState.stormClusterState.topologyLogConfig(topologyId, null);
+        logConfigManager.processLogConfigChange(logConfig);
+        establishLogSettingCallback();
+    }
+
+    public void establishLogSettingCallback() {
+        workerState.stormClusterState.topologyLogConfig(topologyId, this::checkLogConfigChanged);
+    }
+
+
+    /**
+     * make a handler for the worker's send disruptor queue to
+     * check highWaterMark and lowWaterMark for backpressure
+     */
+    private DisruptorBackpressureCallback mkDisruptorBackpressureHandler(WorkerState workerState) {
+        return new DisruptorBackpressureCallback() {
+            @Override public void highWaterMark() throws Exception {
+                workerState.transferBackpressure.set(true);
```

Comment:
You are correct that was a bug that was fixed, but the logs were not fixed.  I'll update them to be correct.